packages/design-system/src/components/tree/tree.tsx

```diff
@@ -44,6 +44,8 @@ export type TreeProps<Data extends { id: string }> = {
     dropTarget: { itemSelector: ItemSelector; position: number | "end" };
   }) => void;
   onCancel: () => void;
+  isEditingItemName: ItemId | undefined;
```

Comment:
confusing naming, you are passing an "id", but call property a "name", also isSomething is always boolean

suggestion: `editingItemId`